src/asci.cc

```diff
@@ -0,0 +1,1018 @@
+/*
+ * @BEGIN LICENSE
+ *
+ * Forte: an open-source plugin to Psi4 (https://github.com/psi4/psi4)
+ * that implements a variety of quantum chemistry methods for strongly
+ * correlated electrons.
+ *
+ * Copyright (c) 2012-2017 by its authors (see COPYING, COPYING.LESSER,
+ * AUTHORS).
+ *
+ * The copyrights for code used from other parties are included in
+ * the corresponding files.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/.
+ *
+ * @END LICENSE
+ */
+
+#include "psi4/libpsi4util/process.h"
+#include "psi4/libmints/molecule.h"
+#include "psi4/libmints/pointgrp.h"
+#include "psi4/libpsio/psio.hpp"
+
+#include "../helpers/printing.h"
+#include "asci.h"
+
+using namespace psi;
+
+namespace psi {
+namespace forte {
+
+//#ifdef _OPENMP
+//#include <omp.h>
+//#else
+//#define omp_get_max_threads() 1
+//#define omp_get_thread_num() 0
+//#define omp_get_num_threads() 1
+//#endif
+
+void set_ASCI_options(ForteOptions& foptions) {
+    /* Convergence Threshold -*/
+    foptions.add_double("ASCI_CONVERGENCE", 1e-5, "ASCI Convergence threshold");
```

Comment:
Is this an energy convergence? If yes, follow the psi convention and call it asci_e_convergence to differentiate from density (d_) or residual (r_).

Reply:
Yes it is energy, I'll update the option name.